### ISSUE TITLE
Rename `get_n_species()` to `n_species()`

### ISSUE THREAD

**Issue by peterdesmet** [inbo/camtraptor#243]
_Suggested in camtraptor July 2023 coding sprint_

Rationale: consistency. Some parameters should be removed as well

```R
n_species(
  package = NULL,
  group_by = "deploymentID" # new param, not sure if other groups make sense
  # ... removed, see filters
  # datapkg removed
)
```

The returned information would be:

```ini
deploymentID
n_species # currently n
```

**Comment by damianooldoni** [comment]
About deprecating vs removing old function: see https://github.com/inbo/camtraptor/issues/237#issuecomment-2247772945

**Comment by damianooldoni** [comment]
Solved by PR #331.
